src/storage/cookie.zig

```diff
@@ -0,0 +1,922 @@
+                c.deinit();
+                if (is_expired) {
+                    _ = self.cookies.swapRemove(i);
+                } else {
+                    self.cookies.items[i] = cookie;
+                }
+                return;
+            }
+        }
+
+        if (!is_expired) {
+            try self.cookies.append(self.allocator, cookie);
+        }
+    }
+
+    pub fn forRequest(
+        self: *Jar,
+        allocator: Allocator,
+        request_time: i64,
+        origin_uri: ?Uri,
+        target_uri: Uri,
+        navitation: bool,
+    ) !CookieList {
+        const target_path = target_uri.path.percent_encoded;
+        const target_host = (target_uri.host orelse return error.InvalidURI).percent_encoded;
+
+        const same_site = try areSameSite(origin_uri, target_host);
+        const is_secure = std.mem.eql(u8, target_uri.scheme, "https");
+
+        var matching: std.ArrayListUnmanaged(*Cookie) = .{};
+
+        var i: usize = 0;
+        var cookies = self.cookies.items;
+        while (i < cookies.len) {
+            const cookie = &cookies[i];
+
+            if (isCookieExpired(cookie, request_time)) {
+                cookie.deinit();
+                _ = self.cookies.swapRemove(i);
+                // don't increment i !
+                continue;
+            }
+            i += 1;
+
+            if (is_secure == false and cookie.secure) {
+                // secure cookie can only be sent over HTTPs
+                continue;
+            }
+
+            if (same_site == false) {
+                // If we aren't on the "same site" (matching 2nd level domain
+                // taking into account public suffix list), then the cookie
+                // can only be sent if cookie.same_site == .none, or if
+                // we're navigating to (as opposed to, say, loading an image)
+                // and cookie.same_site == .lax
+                switch (cookie.same_site) {
+                    .strict => continue,
+                    .lax => if (navitation == false) continue,
+                    .none => {},
+                }
+            }
+
+            {
+                const domain = cookie.domain;
+                if (domain[0] == '.') {
+                    // when explicitly set, the domain
+                    //  1 - always starts with a .
+                    //  2 - always is a suffix match (or examlpe)
+                    if (std.mem.eql(u8, target_host, domain[1..]) == false and std.mem.endsWith(u8, target_host, domain) == false) {
+                        continue;
+                    }
+                } else if (std.mem.eql(u8, target_host, domain) == false) {
+                    // when Domain=XYX isn't specified, it's an exact match only
+                    continue;
+                }
+            }
+
+            {
+                const path = cookie.path;
+                if (path[path.len - 1] == '/') {
+                    // If our cookie path is doc/
```

Comment:
You mean `/doc/`? The parser ensure the path starts with a `/` isn't it?

Reply:
Cleaned up the comment